integration/setup_test.go

```diff
@@ -31,22 +34,52 @@ func (s *integrationTestSuite) Setup() {
 		o := obj
 		// check if object already exists
 		var existingObj client.Object
-		_ = integration.Client.Get(ctx, client.ObjectKey{
+		switch {
+		case obj.GroupVersionKind().Kind == "Namespace":
+			existingObj = &corev1.Namespace{}
+		case obj.GroupVersionKind().Kind == "CustomResourceDefinition":
+			existingObj = &apiextensionsv1.CustomResourceDefinition{}
+		case obj.GroupVersionKind().Kind == "Deployment":
+			existingObj = &appsv1.Deployment{}
+		case obj.GroupVersionKind().Kind == "ServiceAccount":
+			existingObj = &corev1.ServiceAccount{}
+		case obj.GroupVersionKind().Kind == "Role":
+			existingObj = &rbacv1.Role{}
+		case obj.GroupVersionKind().Kind == "RoleBinding":
+			existingObj = &rbacv1.RoleBinding{}
+		case obj.GroupVersionKind().Kind == "ClusterRole":
+			existingObj = &rbacv1.ClusterRole{}
+		case obj.GroupVersionKind().Kind == "ClusterRoleBinding":
+			existingObj = &rbacv1.ClusterRoleBinding{}
+		case obj.GroupVersionKind().Kind == "Service":
+			existingObj = &corev1.Service{}
+		case obj.GroupVersionKind().Kind == "Secret":
+			existingObj = &corev1.Secret{}
+		case obj.GroupVersionKind().Kind == "ValidatingWebhookConfiguration":
+			existingObj = &admissionv1.ValidatingWebhookConfiguration{}
+		case obj.GroupVersionKind().Kind == "ServiceMonitor":
+			existingObj = &monitoringv1.ServiceMonitor{}
+		default:
+			s.T().Fatalf("not supported kind object %v %v %v", o.GroupVersionKind(), o.GetNamespace(), o.GetName())
+		}
+		err := integration.Client.Get(ctx, client.ObjectKey{
 			Namespace: o.GetNamespace(),
 			Name:      o.GetName(),
 		}, existingObj)
 
-		if existingObj == nil {
+		if err != nil && errors.IsNotFound(err) {
+			s.T().Log("error not found:", err, o.GroupVersionKind(), o.GetNamespace(), o.GetName())
 			// if not create one
-			err := integration.Client.Create(ctx, &o)
+			err = integration.Client.Create(ctx, &o)
 			s.Require().NoError(err)
 
-			s.T().Log("created: ", o.GroupVersionKind().String(),
-				o.GetNamespace()+"/"+o.GetName())
+			s.T().Log("created:", o.GroupVersionKind(), o.GetNamespace(), o.GetName())
 
 			if o.GetKind() == "Deployment" {
 				deployments = append(deployments, o)
 			}
+		} else {
+			s.T().Log("found:", err, o.GroupVersionKind(), o.GetNamespace(), o.GetName())
```

Comment:
Do we need to have `err` here? Seems like it will always be `nil`.

Do you mind adding a more verbose message such as `"found <GVK>/<NS>/<Name>; skipping creation..."`